tests/integration_tests.rs

```diff
@@ -6,12 +6,172 @@ mod install {
     use std::process::Command;
     use test_case::test_case;
 
-    use crate::fs::{any_temp_dir, path_to_string};
+    use crate::fs::{any_temp_dir, any_temp_file, path_to_string};
     use assert_cmd::assert::OutputAssertExt;
     use assert_cmd::prelude::CommandCargoExt;
 
     use crate::assertions::assert_file_exists;
 
+    #[test]
+    fn destination_is_not_dir() {
```

Comment:
Remove this test or transform it in a unit test in `download_handler.rs`.

End to end tests are expensive to write and maintain. In this case, this one doesn't add much value.

This could be a unit test on `download_handler.rs` on the `destination_may_not_dir` function.